commons/subprocess/instance.go

```diff
@@ -81,15 +87,112 @@ func (s *Instance) Close() error {
 	return err
 }
 
+// UserInfo has user info
+type UserInfo struct {
+       UID uint32
+       GID uint32
+}
+
+func isMemeberOf(u *osuser.User, group *osuser.Group) (bool, error) {
+
+	groupids, err := u.GroupIds()
+	if err != nil {
+		return false, err
+	}
+	for _, gid := range groupids {
+		if gid == group.Gid {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// NewUserInfo sets up user
+func NewUserInfo(u string) (*UserInfo, error) {
+	var username, usergroup string
+	var gid int
+
+	input := strings.Split(u, ":")
+
+	switch len(input) {
+	case 1:
+		username = strings.Trim(input[0], " ")
+	case 2:
+		username = strings.Trim(input[0], " ")
+		usergroup = strings.Trim(input[1], " ")
+	default:
+		return nil, errors.New("Incorrect user or user:group data")
+	}
+
+	user, err := osuser.Lookup(username)
+	if err != nil {
+		return nil, err
+	}
+
+	uid, err := strconv.Atoi(user.Uid)
+	if err != nil {
+		return nil, err
+	}
+
+	if usergroup != "" {
+		group, err := osuser.LookupGroup(usergroup)
+		if err != nil {
+			return nil, err
+		}
+		isMember, err := isMemeberOf(user, group)
+		if err != nil {
+			return nil, err
+		}
+		if isMember {
+			gid, err = strconv.Atoi(group.Gid)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			errorMessage := fmt.Sprintf("user %s is not a member of %s group\n", username, usergroup)
+			return nil, errors.New(errorMessage)
+		}
+	} else {
+		groupids, err := user.GroupIds()
+		if err != nil {
+			return nil, err
+		}
+
+		gid, err = strconv.Atoi(groupids[0])
+			if err != nil {
+				return nil, err
+			}
+		}
+
+	return &UserInfo{UID: uint32(uid), GID: uint32(gid)}, nil
+}
```

Comment:
Although the `Instance` type depends on the `UserInfo` type, they're separate things. Could you move this code into a separate module (i.e. file), e.g. `userinfo.go`?  Being in a separate module would mean `NewUserInfo` is renamed to `New` and `instance.go` would call it like `userinfo.New(...)`.